sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -99,6 +99,7 @@ case class AdaptiveSparkPlanExec(
     // Skew join does not handle `CustomShuffleReader` so needs to be applied first.
     OptimizeSkewedJoin,
     CoalesceShufflePartitions(context.session),
+    ExpandShufflePartitions,
```

Comment:
shall we do it before `CoalesceShufflePartitions`? to follow `OptimizeSkewedJoin`

Reply:
yea,  seems better